library/ssl_tls13_client.c

```diff
@@ -2095,6 +2095,19 @@ int mbedtls_ssl_get_early_data_status( mbedtls_ssl_context *ssl )
 
     return( ssl->early_data_status );
 }
+
+void mbedtls_ssl_set_early_data( mbedtls_ssl_context *ssl,
+                                 char *buffer, unsigned int len )
+{
+    if( ssl != NULL )
+    {
+        if( buffer != NULL && len > 0 )
```

Comment:
If the expectation is that users will pass a non NULL buffer, then we should return `BAD_INPUT_DATA` here if `buffer == NULL`. 

Is there any use in having the client negotiated 0-RTT but then actually nothing, or empty 0-RTT records? If so, we would need to reflect this usecase in the API.